src/main/java/ch/njol/skript/lang/function/Functions.java

```diff
@@ -268,32 +224,59 @@ private static Signature<?> signError(String error) {
 
 	/**
 	 * Gets a function, if it exists. Note that even if function exists in scripts,
-	 * it might not have been parsed yet. If you want to check for existance,
-	 * then use {@link #getSignature(String)}.
-	 * @param name Name of function.
+	 * it might not have been parsed yet. If you want to check for existence,
+	 * then use {@link #getSignature(String, String, boolean)}.
+	 *
+	 * @param name   Name of function.
+	 * @param globalOnly Whether it should ignore local functions. Otherwise, local functions are prioritized.
 	 * @return Function, or null if it does not exist.
 	 */
 	@Nullable
-	public static Function<?> getFunction(String name) {
-		Namespace namespace = globalFunctions.get(name);
-		if (namespace == null) {
-			return null;
+	public static Function<?> getFunction(String name, @Nullable String script, boolean globalOnly) {
+		Namespace namespace = null;
+		Function<?> function = null;
+		if (script != null && !globalOnly) {
+			namespace = getScriptNamespace(script);
+			if (namespace != null)
+				function = namespace.getFunction(name);
+		}
+		if (namespace == null || function == null) {
+			namespace = globalFunctions.get(name);
+			if (namespace == null)
+				return null;
+			function = namespace.getFunction(name, false);
 		}
-		return namespace.getFunction(name);
+		return function;
 	}
 
 	/**
 	 * Gets a signature of function with given name.
 	 * @param name Name of function.
+	 * @param globalOnly Whether it should ignore local functions. Otherwise, local functions are prioritized.
 	 * @return Signature, or null if function does not exist.
 	 */
 	@Nullable
-	public static Signature<?> getSignature(String name) {
-		Namespace namespace = globalFunctions.get(name);
-		if (namespace == null) {
-			return null;
+	public static Signature<?> getSignature(String name, @Nullable String script, boolean globalOnly) {
```

Comment:
Same as above.

I think a private internal method should be made to obtain the namespace (since these methods are pretty similar)